pkg/gateway/main_test.go

```diff
@@ -0,0 +1,133 @@
+package gateway_test
+
+import (
+	"context"
+	"io"
+	"log"
+	"net/http"
+	"net/http/httptest"
+	"os"
+	"testing"
+
+	"github.com/ory/dockertest/v3"
+	"github.com/spf13/viper"
+	"github.com/treeverse/lakefs/pkg/block"
+	"github.com/treeverse/lakefs/pkg/catalog"
+	"github.com/treeverse/lakefs/pkg/config"
+	"github.com/treeverse/lakefs/pkg/gateway"
+	"github.com/treeverse/lakefs/pkg/gateway/multiparts"
+	"github.com/treeverse/lakefs/pkg/gateway/simulator"
+	"github.com/treeverse/lakefs/pkg/kv"
+	"github.com/treeverse/lakefs/pkg/kv/kvtest"
+	_ "github.com/treeverse/lakefs/pkg/kv/mem"
```

Comment:
See my previous response:

> I like the approach where unit tests use 'mem' and system tests use postgres. I think the only reason we used postgres in our unit tests is because we didn't have any other option.
We can take advantage of the new mem implementation and also save a lot of runtime. WDYT?

Reply:
Since this is a unit test and we are testing the tracker functionality and not the KV functionality I believe the underlying KV implementation is irrelevant. In any case - the postgres implementation should be tested by it's own in a dedicated unit test and will be tested extensively as part of the system tests